ceno_emul/src/rv32im.rs

```diff
@@ -270,44 +270,25 @@ impl DecodedInstruction {
         }
     }
 
-    /// The internal view of the immediate, for use in circuits.
-    pub fn imm_internal(&self) -> u32 {
+    pub fn codes(&self) -> InsnCodes {
+        FastDecodeTable::get().lookup(self)
+    }
+
+    pub fn immediate_msb(&self) -> u32 {
```

Comment:
One reason to have getters instead of direct access to public struct member is to do error handling.

This getter is only valid for instructions that have an immediate field.  (And even for those, it only makes sense for immediate shifts, I think?)  But the getter here is happy to silently hand you garbage data.

As a minimal change, you could at least make it return a consistent 0 for instructions that don't have an immediate field in the spec.

(But more holistically, we should probably move to `rrs-lib` for decoding like SP1 does.)

Reply:
Removed in #546.